ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -27,44 +37,62 @@ pub unsafe trait RefCounted where Self: Sized {
         };
 
         let result = cb(owned.as_ref());
+        // this isn't unwind safe, do not use without panic=abort
         forget(owned);
 
         result
     }
 
+    /// Given a non-null strong FFI reference, converts it into an Arc.
+    /// Panics on null.
```

Comment:
Why not return an Option<Arc<Self>> and let the caller choose whether or not to panic with unwrap()?